src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -275,6 +285,11 @@ define(function (require) {
         return '' + self.toJSON();
       };
 
+      self.destroy = function () {
+        patternCache.clear(self.id);
```

Comment:
Same as above - exception if id is not set.  It probably goes without saying that these exceptions should be tested as well.